### PR TITLE
add UI for snapshots (dev option)

### PR DESCRIPTION
**Pull request by andreievg**
Requires back end branch in this PR: https://github.com/openmsupply/application-manager-server/pull/402

New dev option to allowing to use, take, upload, and download snapshots (the diff option is missing, not a priority).

Since this is a dev feature, I am ok with inline styles remaining

### CODE REVIEW

File: src/containers/Dev/Snapshots.tsx
Comment (CarlosNZ) [inline src/containers/Dev/Snapshots.tsx:52]: Maybe replace with "_" rather than empty string -- so if user types "my snapshot", they'll get `my_snapshot` rather than `mysnapshot` ?

File: src/containers/Dev/Snapshots.tsx
Comment (CarlosNZ) [inline src/containers/Dev/Snapshots.tsx:69]: If there is an error, or jsonResult.success === false, will the Portal ever close?

File: src/containers/Dev/Snapshots.tsx
Comment (CarlosNZ) [inline src/containers/Dev/Snapshots.tsx:113]: Typo: `renderSnapshotList`

File: src/containers/Dev/Snapshots.tsx
Comment (CarlosNZ) [inline src/containers/Dev/Snapshots.tsx:55]: The back-end handles no parameter and sets a default "current" -- so we should allow it to be run with no input name.

File: src/containers/Dev/Snapshots.tsx
Comment (CarlosNZ) [inline src/containers/Dev/Snapshots.tsx:257]: Typos x2

**Review by CarlosNZ — CHANGES_REQUESTED** [review]

File: src/containers/Dev/Snapshots.tsx
Comment (andreievg) [reply src/containers/Dev/Snapshots.tsx:47]: 1a95b7e7e9ff9e940f500b5c8e299fd5d78e8b12

File: src/containers/Dev/Snapshots.tsx
Comment (andreievg) [reply src/containers/Dev/Snapshots.tsx]: 129f8029975a8a4016eca2dbc63ba06db0a06bcb

File: src/containers/Dev/Snapshots.tsx
Comment (andreievg) [reply src/containers/Dev/Snapshots.tsx]: had to manually click on the close button in the portal error, now it's changed to be a modal, should close as per normal modal closing (or when clicking on 'X' in error label)

File: src/containers/Dev/Snapshots.tsx
Comment (andreievg) [reply src/containers/Dev/Snapshots.tsx:247]: wow soo many typos, and even the branch name lol

552e1ef8c8a37bfa35e25b070eae62fd326b5df2

File: src/containers/Dev/Snapshots.tsx
Comment (andreievg) [reply src/containers/Dev/Snapshots.tsx:55]: I thought it was a bit weird to takeSnapshot with no name, and have it create/overwrite the one named 'current' ?

File: src/containers/Dev/Snapshots.tsx
Comment (andreievg) [reply src/containers/Dev/Snapshots.tsx]: 552e1ef

File: src/containers/Dev/Snapshots.tsx
Comment (CarlosNZ) [reply src/containers/Dev/Snapshots.tsx:55]: Sure, then maybe the front-end UI could insert a default name in the input field?

Comment (andreievg) [reply src/containers/Dev/Snapshots.tsx:55]: could do, current placeholder is 'New snapshot' to indicate what input is related to, and if we do default value it would appear instead of placeholder, so not sure..

**Comment by nmadruga** [comment]
Would be great to have this PR included at some point so we can do snapshots on the `master` branch. Any plans for merging this soon @andreievg ?

**Comment by andreievg** [comment]
> Would be great to have this PR included at some point so we can do snapshots on the master branch. Any plans for merging this soon @andreievg ?

I am keen to merge just needs a re-review by @CarlosNZ

**Comment by CarlosNZ** [comment]
> > Would be great to have this PR included at some point so we can do snapshots on the master branch. Any plans for merging this soon @andreievg ?
> 
> I am keen to merge just needs a re-review by @CarlosNZ

Sorry, didn't realise this was still hanging out there.

**Review by CarlosNZ — APPROVED** [review]

**Comment by CarlosNZ** [comment]
One quick thing before you merge, can you re-run `yarn generate` so the graphql definitions file gets updated, noticed it changed after re-running on mine.

**Comment by andreievg** [comment]
> One quick thing before you merge, can you re-run yarn generate so the graphql definitions file gets updated, noticed it changed after re-running on mine.

done, thanks

**Pull request merged**
